include/seastar/coroutine/generator.hh

```diff
@@ -26,6 +26,13 @@
 #include <utility>
 #include <seastar/core/future.hh>
 
+
+#ifndef SEASTAR_COROUTINES_ENABLED
+#error Coroutines support disabled.
+#else
+#include <coroutine>
+#endif
```

Comment:
i'd suggest just
```c++
#include <coroutine>
```

as the error message would be self-explained.

Reply:
or, if we want to mirror the way how `coroutine.hh` checks for the availability using `std-compat.hh`, could put
```c++
#include <seastar/util/std-compat.hh>
#ifndef SEASTAR_COROUTINES_ENABLED
#error Coroutines support disabled.
#endif
#include <coroutine>
```
as the macro is offered by `seastar/util/std-compat.hh`, we should not rely on ``seastar/core/future.h` to do this for us. but ... personally, i still prefer `#include <coroutine>` directly. as we don't have "error handling" here.

Reply:
Updated